sesam.py

```diff
@@ -1182,12 +1182,53 @@ def authenticate(self):
         else:
             pass
 
+    def check_template_sink(self):
+        """
+        Checks the pipe templates for `sink.dataset`, if it has a value we exit with a
+        warning. (See task IS-14872)
+
+        Returns:
+            Boolean: True - If all is good.
+        """
+        connector_dir = os.getcwd()
+        with open(f"{connector_dir}/manifest.json", "r") as mf:
+            manifest = json.load(mf)
+            template_files = [
+                manifest.get("datatypes").get(tf).get("template")
+                for tf in manifest.get("datatypes")
+            ]
+
+        files_with_warnings = list()
+        for template_file in template_files:
+            with open(f"{connector_dir}/{template_file}", "r") as tf:
+                template = json.load(tf)
+
+            if type(template) is not list:
+                continue
+
+            for obj in template:
+                if obj.get("type") == "pipe" and obj.get("sink", {}).get("dataset"):
+                    files_with_warnings.append(
+                        f"[!] Sink.dataset should not have a value in {template_file}. "
+                        "Please remove the value to continue."
+                    )
+
+        if len(files_with_warnings) > 0:
+            for warning in files_with_warnings:
+                logger.error(warning)
+            sys.exit(1)
+
+        return True
+
     def validate(self):
         logger.info("Validating config files")
         # set the current directory when sesam validate is called from root.
         if self.args.command == "validate" and self.args.connector_dir != ".":
             os.chdir(self.args.connector_dir)
+
         is_valid = True
+        self.check_template_sink()
```

Comment:
I'd set is_valid to true or false depending on how `check_template_sink` turns out and let the exit logic at the end of the validate method deal with the rest.